### PR TITLE
feat: DataViz - support Activity Flows on the Summary tab (M2-6537)

### PR DESCRIPTION
**Pull request by IhorNesterenko99**
- [x] Tests for the changes have been added
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-5559](https://mindlogger.atlassian.net/browse/M2-5559) - parent ticket (FE and BE)
🔗 [Jira Ticket M2-6537](https://mindlogger.atlassian.net/browse/M2-6537)

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/ScatterChart/ChartTooltip/ChartTooltip.utils.test.tsx [hunk 1]
```diff
@@ -5,7 +5,12 @@ import { getReviewOption } from './ChartTooltip.utils';
 describe('getReviewOption', () => {
   test.each([
     // [mine, other, expected]
-    [0, 0, null, 'returns null when both mine and other are 0'],
+    [
+      0,
+      0,
+      t('leaveReview'),
+      'returns correct translation for leave review when both mine and other are 0',
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/ScatterChart/ChartTooltip/ChartTooltip.utils.test.tsx]: let's move 4th arguments on first place as `description` for test cases and change the row 32 accordingly:
`])('correctly processes mine: %i, other: %i', (mine, other, expected) => {` 
=> 
`])('%d', (_description, mine, other, expected) => {`

**UPD:**
or remove at all the 4th argument in all cases as redundant

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx [hunk 2]
```diff
@@ -1,120 +1,105 @@
...
     responseOptions,
     subscalesFrequency,
-    selectedActivity,
+    selectedEntity,
     identifiers,
     apiVersions,
     versions,
+    flowSubmissions,
+    flowResponses,
   ]: [
```
Comment (nmialik) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx]: lets maybe move this type to file with types?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/DownloadReport/DownloadReport.hooks.test.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+      const currentResult = result.current as ReturnType<typeof useDownloadReport>;
+      expect(currentResult.isDownloadReportDisabled).toBe(expected);
+      expect(currentResult.isDownloadReportLoading).toBe(false);
+      expect(currentResult.downloadReportError).toBe(null);
+    },
+  );
+
+  test('handles download report for Activity', async () => {
+    // const mockedDownload = jest.spyOn(downloadJs, 'default').mockImplementation(() => {});
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/DownloadReport/DownloadReport.hooks.test.ts]: please remove the comment

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts [hunk 4]
```diff
@@ -48,6 +48,7 @@ const testIdentifierDatesChange = async () => {
           respondentId: mockedRespondentId,
           toDatetime: '2024-04-12T23:59:00',
           versions: 'v3',
+          limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts]:
```suggestion
          limit: MAX_LIMIT,
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts [hunk 5]
```diff
@@ -3,8 +3,8 @@ import mockAxios from 'jest-mock-axios';
 import { endOfDay, startOfDay, subDays } from 'date-fns';
 
 import { mockedActivityId, mockedAppletId, mockedRespondentId } from 'shared/mock';
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts:5]:
```suggestion
import { mockedActivityId, mockedAppletId, mockedRespondentId } from 'shared/mock';
import { MAX_LIMIT } from 'shared/consts'; 
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts [hunk 6]
```diff
@@ -151,6 +270,7 @@ describe('useRespondentAnswers', () => {
             respondentId: mockedRespondentId,
             toDatetime: '2024-01-15T17:00:00',
             versions: 'v3',
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts]:
```suggestion
            limit: MAX_LIMIT,
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts [hunk 7]
```diff
@@ -165,6 +285,80 @@ describe('useRespondentAnswers', () => {
...
+        `/answers/applet/${mockedAppletId}/flows/${mockedFlowId}/submissions`,
+        {
+          params: {
+            emptyIdentifiers: true,
+            fromDatetime: '2024-01-12T09:00:00',
+            identifiers: undefined,
+            respondentId: mockedRespondentId,
+            toDatetime: '2024-01-15T17:00:00',
+            versions: 'v3',
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts]:
```suggestion
            limit: MAX_LIMIT,
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts [hunk 8]
```diff
@@ -208,6 +402,7 @@ describe('useRespondentAnswers', () => {
             respondentId: mockedRespondentId,
             toDatetime: '2024-01-18T23:59:00',
             versions: 'v3',
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.test.ts]:
```suggestion
            limit: MAX_LIMIT,
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/ScatterChart/ChartTooltip/ChartTooltip.test.tsx [hunk 9]
```diff
@@ -9,22 +7,27 @@ import { page } from 'resources';
...
 const routePath = page.appletRespondentDataSummary;
 
 const dataTestid = 'scatter-chart';
-const answerId = '773d904e-15a0-4702-b53b-d3f3e2d8be71';
+const answerId = 'some-answer-id';
+const flowSubmitId = 'some-submit-id';
 const date = 1703089235000; // Dec 20 2023, 16:20:35
 
-const getProps = (areSubscalesVisible = false) => ({
+// const getProps = (areSubscalesVisible = false) => ({
```
Comment (nmialik) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/ScatterChart/ChartTooltip/ChartTooltip.test.tsx]: please remove the comment

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportHeader/ReportHeader.tsx [hunk 10]
```diff
@@ -1,45 +1,27 @@
...
-        </Tooltip>
-        {error && <StyledErrorText sx={{ mt: theme.spacing(0.8) }}>{error}</StyledErrorText>}
-      </Box>
+    <StyledHeader data-testid={dataTestId} isSticky={isHeaderSticky}>
+      {name && (
+        <StyledStickyHeadline isSticky={isHeaderSticky} color={variables.palette.on_surface}>
+          {name}
+        </StyledStickyHeadline>
+      )}
+      {/*TODO: implement Download Latest Report (Combined) for Flow after back-end is ready*/}
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportHeader/ReportHeader.tsx]: let's add the task number

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.test.tsx [hunk 11]
```diff
@@ -175,12 +183,42 @@ describe('ReportFilters', () => {
...
         {
           params: {
-            emptyIdentifiers: true,
+            emptyIdentifiers: false,
+            fromDatetime: '2024-01-04T00:00:00',
+            identifiers: '',
+            respondentId: mockedRespondentId,
+            toDatetime: '2024-01-10T23:59:00',
+            versions: '1.0.0,1.0.1',
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.test.tsx]: let's use MAX_LIMIT

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.test.tsx [hunk 12]
```diff
@@ -175,12 +183,42 @@ describe('ReportFilters', () => {
...
+        `/answers/applet/${mockedAppletId}/flows/${mockedFlow.id}/submissions`,
+        {
+          params: {
+            emptyIdentifiers: false,
             fromDatetime: '2024-01-04T00:00:00',
             identifiers: '',
             respondentId: mockedRespondentId,
             toDatetime: '2024-01-10T23:59:00',
             versions: '1.0.0,1.0.1',
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.test.tsx]: let's use MAX_LIMIT

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx [hunk 13]
```diff
@@ -1,120 +1,105 @@
...
+  const { isFlow, hasAnswer } = selectedEntity;
 
-    if (data) {
-      const contentDisposition = headers?.['content-disposition'];
-      const fileName =
-        (contentDisposition && LATEST_REPORT_REGEX.exec(contentDisposition)?.groups?.filename) ??
-        LATEST_REPORT_DEFAULT_NAME;
-      const base64Str = Buffer.from(data).toString('base64');
-      const linkSource = getLatestReportUrl(base64Str);
+  // TODO: check subscales display for Flows, add reviewCount for Flows when back-end is ready
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx]: let's link/add the task number

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 14]
```diff
@@ -1,13 +1,22 @@
...
+  flowSubmissions,
+  answers,
+}: GetCompletions): Completion[] =>
+  isFlow
+    ? flowSubmissions.map(({ submitId, endDatetime, createdAt }) => ({
+        id: submitId,
+        endDatetime: endDatetime ?? createdAt,
+        areSubscalesVisible: false,
+        isFlow,
+        // TODO: add review count support for Flows
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts]: do we have a task for it or a dependency with backend task?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.test.tsx [hunk 15]
```diff
@@ -147,6 +168,17 @@ describe('RespondentDataSummary component', () => {
     await waitFor(() => {
       expect(mockAxios.get).toHaveBeenNthCalledWith(
         1,
+        `/answers/applet/${mockedAppletId}/summary/flows`,
+        {
+          params: {
+            limit: 10000,
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.test.tsx]: let's use MAX_LIMIT

**Review by nmialik — APPROVED** [review]

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**